### ISSUE TITLE
Need a pattern matcher for identifying empty raises with rescue or resbody as ancestors or relevant documentation 

### ISSUE THREAD

**Issue by adip1818** [rubocop/rubocop#10901]
``` RUBY
 def_node_matcher :rescue_raises?, <<~PATTERN
          (resbody ... `(send nil? :raise))
        PATTERN
```
Using this pattern-matcher to identify rescue/resbody nodes, having empty raise node as a descendent.
Need a pattern-matcher to identify empty raise nodes having rescue as an ancestor(at any level) or relevant documents or reference material.``

### Empty raise within a rescue 
``` RUBY
                begin
                  this_will_fail!
                rescue Failure => error
                  log.error error.message
                  raise
                 end
```
### Example node structure
(resbody
  (array
    (const nil :Failure))
  (lvasgn :error)
  (begin
    (send
      (send nil :log) :error
      (send
        (lvar :error) :message))
    (send nil :raise)))

**Comment by jonas054** [comment]
The documentation for the pattern language is [here](https://docs.rubocop.org/rubocop-ast/node_pattern.html). Your pattern definition looks correct to me. You could add a `$` to capture the node you want to highlight in the offense report.

Example cop code:
```ruby
# frozen_string_literal: true

module RuboCop
  module Cop
    module SomeDepartment
      # Documentation...
      class EmptyRaise < Base
        MSG = 'Empty raise detected'

        # @!method rescue_raises?(node)
        def_node_matcher :rescue_raises?, <<~PATTERN
          (resbody ... `$(send nil? :raise))
        PATTERN

        def on_resbody(node)
          raise_node = rescue_raises?(node)
          add_offense(raise_node) if raise_node
        end
      end
    end
  end
end
```
